src/main/kotlin/org/rust/cargo/project/model/impl/CargoProjectImpl.kt

```diff
@@ -592,7 +592,7 @@ data class CargoProjectImpl(
             workspaceStatus = UpdateStatus.UpToDate,
             userDisabledFeatures = userDisabledFeatures.retain(result.value.packages)
         )
-        is TaskResult.Err -> copy(workspaceStatus = UpdateStatus.UpdateFailed(result.reason))
+        is TaskResult.Err -> copy(workspaceStatus = UpdateStatus.UpdateFailed(result.message ?: result.reason))
```

Comment:
`result.message` is not intended to be shown here. Is is shown in the `Sync` tab of the `Build` tool window.

Reply:
Okay, interesting.

I made this change so that errors when running Cargo would be logged when running the tests. Without this change, my failing test just logged `"Failed to run Cargo"`, and did not include the command line or anything else to be sure that it was the `cargo metadata` call failing as described in bug #10125.

The reason for this is, if `message` is not included in the TaskResult here, the Err value returned from [line 387 of `CargoSyncTask.fetchCargoWorkspace()`](https://github.com/intellij-rust/intellij-rust/blob/6f5888b353ae2757b19ee5e6db5d79d55d7823b1/src/main/kotlin/org/rust/cargo/project/model/impl/CargoSyncTask.kt#L387) gets converted to `UpdateFailed("Failed to run Cargo")`, and the `message` value with the actual error details is not included. This means the testing code doesn't log it in case of failure.

Is there a better way to fix that issue in the tests without changing the output of the tabs in the GUI? Maybe there should be some logging in that `unwrapOrElse` block in `fetchCargoWorkspace()`, like the other ones in that method?

If not, happy to revert this change. In my case, I just had to step through with the debugger to confirm my test was causing the correct underlying error, so I thought it would be an improvement.

Reply:
I've reverted this for now. Happy to implement another solution if you can suggest one, otherwise all the feedback has been addressed.